cmd/kubeadm/app/preflight/checks.go

```diff
@@ -912,25 +912,58 @@ func RunInitNodeChecks(execer utilsexec.Interface, cfg *kubeadmapi.InitConfigura
 		FileAvailableCheck{Path: kubeadmconstants.GetStaticPodFilepath(kubeadmconstants.Etcd, manifestsDir)},
 		HTTPProxyCheck{Proto: "https", Host: cfg.LocalAPIEndpoint.AdvertiseAddress},
 	}
+
+	// File content check for IPV4 and IPV6 are needed if it is -
+	// (dual stack)  `--service-cidr` or `--pod-network-cidr` is set with a IPV4 and IPV6 cidr, `--apiserver-advertise-address` is optional as it can be auto detected.
+	// (single stack) which is decided by the `--apiserver-advertise-address`.
+	// Note that for the case of dual stack, user might only give IPV6 cidr for `--service-cidr` and leave the `--apiserver-advertise-address` a default value which will be
+	// auto detected and properly bound to a IPV4 address, this will make the cluster non-functional eventually. The case like this should be avoided by the validation instead,
```

Comment:
```suggestion
	// auto detected and properly bound to an IPV4 address, this will make the cluster non-functional eventually. The case like this should be avoided by the validation instead,
```